### ISSUE TITLE
ipywidget interact in qmd file causes quarto to hang without `rmarkdown` installed

### ISSUE THREAD

**Issue by matthew-brett** [quarto-dev/quarto-cli#7274]
### Bug description

Consider the following minimal `index.qmd` page:

~~~rmarkdown
# Title

```{python}
from ipywidgets import interact
```

```{python}
def f(x):
    return x

interact(f, x=10);
~~~

A Quarto project with this file will hang on `quarto render --to html`, whereas the same content in an `.ipynb` file will not hang.

After installing `rmarkdown` in R, for some reason, this build not longer hangs - but a more complex example does hang, even with `rmarkdown` installed.   So I'll leave this report here, in case it is useful to you.  I guess we should not expect the build to hang, even without `rmarkdown` installed.

### Steps to reproduce

```bash
git clone https://github.com/matthew-brett/test-jupyter-widgets
cd test-jupyter-widgets
quarto render --to html
```

### Expected behavior

`quarto render` does not hang and completes project build.

### Actual behavior

`quarto render` hangs indefinitely.

### Your environment

MacOS Big Sur 11.7.10

### Quarto check output

```bash
$ quarto check
Quarto 99.9.9

[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.8: OK
      Dart Sass version 1.55.0: OK
      Deno version 1.33.4: OK

[✓] Checking versions of quarto dependencies......OK

[✓] Checking Quarto installation......OK
      Version: 99.9.9
      Path: /Users/mb312/dev_trees/quarto-cli/package/dist/bin


(|) Checking tools....................
(/) Checking tools....................
(-) Checking tools....................
(\) Checking tools....................
(|) Checking tools....................
(/) Checking tools....................[0m[32mCheck[0m file:///Users/mb312/dev_trees/quarto-cli/src/resources/vendor/deno-land/x/puppeteer@9-0-2/mod.ts

(-) Checking tools....................[2K[G
[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)


(|) Checking LaTeX....................
(/) Checking LaTeX....................
(-) Checking LaTeX....................
(\) Checking LaTeX....................
(|) Checking LaTeX....................
(/) Checking LaTeX....................
(-) Checking LaTeX....................
(\) Checking LaTeX....................[2K[G
[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /Library/TeX/texbin
      Version: 2021


(|) Checking basic markdown render....
(/) Checking basic markdown render....
(-) Checking basic markdown render....
(\) Checking basic markdown render....
(|) Checking basic markdown render....
(/) Checking basic markdown render....
(-) Checking basic markdown render....
(\) Checking basic markdown render....
(|) Checking basic markdown render....
(/) Checking basic markdown render....
(-) Checking basic markdown render....
(\) Checking basic markdown render....
(|) Checking basic markdown render....
(/) Checking basic markdown render....
(-) Checking basic markdown render....
(\) Checking basic markdown render....
(|) Checking basic markdown render....
(/) Checking basic markdown render....
(-) Checking basic markdown render....
(\) Checking basic markdown render....
(|) Checking basic markdown render....
(/) Checking basic markdown render....
(-) Checking basic markdown render....
(\) Checking basic markdown render....
(|) Checking basic markdown render....
(/) Checking basic markdown render....[2K[G
[✓] Checking basic markdown render....OK


(|) Checking Python 3 installation....
(/) Checking Python 3 installation....
(-) Checking Python 3 installation....
(\) Checking Python 3 installation....
(|) Checking Python 3 installation....
(/) Checking Python 3 installation....
(-) Checking Python 3 installation....
(\) Checking Python 3 installation....
(|) Checking Python 3 installation....
(/) Checking Python 3 installation....
(-) Checking Python 3 installation....
(\) Checking Python 3 installation....
(|) Checking Python 3 installation....
(/) Checking Python 3 installation....
(-) Checking Python 3 installation....
(\) Checking Python 3 installation....
(|) Checking Python 3 installation....
(/) Checking Python 3 installation....
(-) Checking Python 3 installation....[2K[G
[✓] Checking Python 3 installation....OK
      Version: 3.11.6
      Path: /Users/mb312/.virtualenvs/probinf/bin/python3
      Jupyter: 5.3.2
      Kernels: python3


(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....
(-) Checking Jupyter engine render....
(\) Checking Jupyter engine render....
(|) Checking Jupyter engine render....
(/) Checking Jupyter engine render....[2K[G
[✓] Checking Jupyter engine render....OK


(|) Checking R installation...........
(/) Checking R installation...........
(-) Checking R installation...........
(\) Checking R installation...........
(|) Checking R installation...........
(/) Checking R installation...........[2K[G
[✓] Checking R installation...........OK
      Version: 4.2.1
      Path: /usr/local/Cellar/r/4.2.1_4/lib/R
      LibPaths:
        - /usr/local/Cellar/r/4.2.1_4/lib/R/library
      knitr: (None)
      rmarkdown: (None)

      The rmarkdown package is not available in this R installation.
      Install with [1minstall.packages("rmarkdown")[22m

```

**Comment by mcanouil** [comment]
I believe this is because you are trying to use an interactive feature that requires a server, see: https://github.com/quarto-dev/quarto-cli/discussions/6496

**Comment by cscheid** [comment]
Thanks for the report. I can't reproduce the hang, unfortunately:

## interact-knitr.qmd

````
---
title: Title
engine: knitr
---

```{python}
from ipywidgets import interact
```

```{python}
def f(x):
    return x

interact(f, x=10)
```
````

## interact-jupyter.qmd

````
---
title: Title
engine: jupyter
---

```{python}
from ipywidgets import interact
```

```{python}
def f(x):
    return x

interact(f, x=10)
```
````

Both of these succeed for me:

```
$ quarto render interact-knitr.qmd --to html

processing file: interact-knitr.qmd
1/4
2/4 [unnamed-chunk-1]
3/4
4/4 [unnamed-chunk-2]
output file: interact.knit.md
...
Output created: interact-knitr.html

$ quarto render interact-jupyter.qmd --to html

Executing 'interact-jupyter.ipynb'
  Cell 1/2: ''...Done
  Cell 2/2: ''...Done
...
Output created: interact-jupyter.html
```

I see you're running 99.9.9. Can you show what commit you're in? I get this:

```
$ git show -s
commit 4c7349406ec38d0dd6f406351ee94fa2dcd2bbb2 (HEAD -> main, origin/main, origin/HEAD)
Merge: 74f07ad7f 89e886fb7
Author: Carlos Scheidegger <285675+cscheid@users.noreply.github.com>
Date:   Wed Oct 18 12:24:46 2023 -0700

    Merge pull request #7271 from quarto-dev/bugfix/7267

    latex: work around longtable counters.
```

**Comment by matthew-brett** [comment]
Hi - just checking - you uninstalled `rmarkdown` from R for the reproduction?

I have quarto-cli commit dcae8ce - from today.

**Comment by matthew-brett** [comment]
I believe I tested with nbclient == 0.8.0, but it's possible I was using 0.7.4 (I would have to uninstall `rmarkdown` myself to check!).

**Comment by cscheid** [comment]
Yes, even though that would make no difference (since your website uses the jupyter engine anyway):

````
R

R version 4.3.1 (2023-06-16) -- "Beagle Scouts"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: aarch64-apple-darwin20 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> remove.packages("rmarkdown")
Removing package from ‘/Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library’
(as ‘lib’ is unspecified)
>
Save workspace image? [y/n/c]: n

$ quarto render

Starting python3 kernel...Done

Executing 'index.ipynb'
  Cell 1/2: ''...Done
  Cell 2/2: ''...Done

pandoc
  to: html
  output-file: index.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en

Output created: _site/index.html
````

**Comment by matthew-brett** [comment]
Yes - I knew that installing `rmarkdown`  should not make a difference.

**Comment by matthew-brett** [comment]
OK - I agree I can't replicate this myself now.   I tried downgrading to nbclient 0.7.4 for a test, got one hang, and then rerunning _in the same directory_ would not start the Jupyter kernel.  Recloning and rerunning worked without problem.  I tried uninstalling R and reinstalling (without any libraries) without effect.   So - sorry for the distraction - there does seem to be something odd going on here, that I can't identify because it is variable - but this report is not a good summary of that - so closing - with thanks for your investigation.
